File: new_cpp/test_crc16.cpp
#include <stdio.h>
#include <iostream>

#include "Crc16.h"
#include "test_data.h"

using namespace std;

unsigned char DATA[90] = TEST_DATA;

int main(int argc, char const *argv[]) {
  cout << "CRC of test data is: " << Crc16(&DATA[0], 90) << endl;
  return 0;
}

File: paperbak/test/test_crc16.py
import unittest

import numpy as np

from paperbak import crc16

from . import TEST_DATA


class TestCRC16(unittest.TestCase):

    def test(self):
        """Test CRC16 over TEST_DATA matches new_cpp/test_crc16.cpp."""
        self.assertEqual(crc16.crc16(bytes(TEST_DATA)), np.uint16(62557))

    def test_type(self):
        """Test CRC16 return correct data type."""
        self.assertEqual(type(crc16.crc16(bytes(TEST_DATA))), np.uint16)


if __name__ == '__main__':
    unittest.main()
